src/comparator/wrappers.py

```diff
@@ -29,25 +29,53 @@
 from google.protobuf import descriptor_pb2
 from google.protobuf.descriptor_pb2 import FieldDescriptorProto
 from src.comparator.resource_database import ResourceDatabase
-from typing import Dict, Sequence, Optional
+from typing import Dict, Sequence, Optional, Tuple
 
 
-# TODO(xiaozhenliu): parse SourceCode location in each descriptor.
+# TODO(xiaozhenliu): parse SourceCode location for properties in each descriptor.
+# For example: during comparison, we will need the source code line number for method.input.
+# The annotations cannot precisely located, because they are customized options, and we
+# use field number to get the location. For extension options, they share the same path [..., 1000].
 @dataclasses.dataclass(frozen=True)
 class EnumValue:
-    """Description of an enum value."""
+    """Description of an enum value.
+
+    proto_file_name: the proto file where the EnumValue exists.
+    source_code_locations: the dictionary that contains all the sourceCodeInfo in the fileDescriptorSet.
+    path: the path to the EnumValue, by querying the above dictionary using the path,
+          we can get the location information.
+    """
 
     enum_value_pb: descriptor_pb2.EnumValueDescriptorProto
+    proto_file_name: str
+    source_code_locations: Dict[Tuple[int, ...], descriptor_pb2.SourceCodeInfo.Location]
+    path: Tuple[int]
 
     def __getattr__(self, name):
         return getattr(self.enum_value_pb, name)
 
+    @property
+    def source_code_line(self):
+        """Return the start line number of source code in the proto file. This is zero-based."""
```

Comment:
Is this correct to state it's zero-based? It looks like the `+ 1` makes this 1-based?

Reply:
sorry for the confusion, updated the comment to make it clearer.

The start line number in `location.span` object is zero-based, and we +1 to get the line number in actual proto file. Same below. Thanks!